testing/src/network/sync.rs

```diff
@@ -172,31 +172,26 @@ async fn block_responder_side() {
     assert_eq!(block, block_struct_1);
 }
 
-#[test]
-#[ignore]
-fn block_propagation() {
-    let rt = tokio::runtime::Builder::new_multi_thread()
-        .enable_io()
-        .enable_time()
-        .build()
-        .unwrap();
+#[tokio::test(flavor = "multi_thread")]
+async fn block_propagation() {
+    let (node, mut peer) = handshaken_node_and_peer(TestSetup::default()).await;
+    let mut peer2 = handshaken_peer(node.local_address().unwrap()).await;
 
-    let setup = TestSetup {
-        consensus_setup: Some(ConsensusSetup {
-            is_miner: true,
-            ..Default::default()
-        }),
-        tokio_handle: Some(rt.handle().clone()),
-        ..Default::default()
-    };
+    let block_1 = BLOCK_1.serialize();
+    let payload = Payload::Block(block_1.clone(), Some(1));
+    peer.write_message(&payload).await;
 
-    rt.block_on(async move {
-        let (_node, mut peer) = handshaken_node_and_peer(setup).await;
+    node.peer_book.broadcast(Payload::Ping(1)).await;
 
-        wait_until!(60, {
-            let payload = peer.read_payload().await.unwrap();
-            matches!(payload, Payload::Block(..))
-        });
+    wait_until!(5, node.storage.canon().await.unwrap().block_height == 1);
+    wait_until!(
+        5,
+        matches!(peer2.read_payload().await.unwrap(), Payload::Block(x, Some(1)) if x == block_1)
+    );
+    wait_until!(5, match peer.read_payload().await.unwrap() {
+        Payload::Block(_, _) => unreachable!(),
+        Payload::Ping(1) => true,
+        _ => false,
```

Comment:
ok, so if I'm not mistaken, we're checking here that `peer2` is able to obtain the `Block` from the `node` (who got it from `peer`), while `peer` is _not_ getting one from  either `node` or `peer2`? sounds good, though we should separately first check that `peer` gets `Ping(1)` and only _then_ we need to ensure that it doesn't receive a `Block` again; this might require the use of a short-lived `tokio::time::timeout` over `peer.read_payload()`, as `wait_until!` requires some breaking condition

Reply:
`peer2` is not connected to `peer` in this case, which should help the test make sense.

Reply:
yep, I checked the default test settings and indeed it should not sync peers :+1:

Reply:
but I forgot that it means the `Ping` from the `Node` won't be delivered, so it should be good as is :ok_hand:

Reply:
it goes beyond that -- the `FakePeer` construction is just a network layer with no logic on top. it can't receive a peer list or connect to anything else.